### ISSUE TITLE
Checkout

### ISSUE THREAD

**Issue by bleroy** [OrchardCMS/OrchardCore.Commerce#16]
Checkout is the process of transforming a shopping cart into an order.

The checkout experience should be based on workflows. The default checkout experience should be defined by a recipe that sets up a workflow, that can then be modified.

The following workflow activities will be necessary:

* Navigate to cart page
* Navigate to shipping page (not all checkout workflows will have a separate shipping page)
* Navigate to payment (not all checkout workflows will have this)
* Set shipping method
* Pay / create order

**Comment by KshitizGIT** [comment]
Hi @bleroy ,

What would the default checkout experience be like using workflows ?
It is something like
GoToCart => Go To Shipping => Payment => Create Order

**Comment by bleroy** [comment]
From the customer's point of view, yes, more or less, and building it as a workflow means it's easy to create variations of that. Don't forget taxes though. From the seller's point of view, that's just the beginning of the workflow of course, but that's a different problem.

**Comment by bleroy** [comment]
To be more specific, we need new activities covering those steps, and we need a setup recipe that creates the default checkout workflow. Then we need the UI that triggers it all, and that accompanies the activities.

**Comment by KshitizGIT** [comment]
Makes sense.. I am trying to build the activities first . But without the shipping implementation, and payment implementation, most of the work will be TBD

**Comment by DemeSzabolcs** [comment]
Hello everyone!

I'm working on integrating Stripe payment, and I noticed that the `order` content type is unfinished. I'm reworking that. 

What should an `order` content item store? It's already storing addresses, that's okay. I'm thinking about statuses like "shipped, arrived, paid, payment failed" etc., and other basic things, like data from the Stripe payment (transaction id) and the ordered products.

What do you think?

**Comment by DemeSzabolcs** [comment]
> status with history of changes

For the history of changes, [Orchard Core Audit Trail](https://docs.orchardcore.net/en/latest/docs/reference/modules/AuditTrail/#audit-trail-orchardcoreaudittrail) seems like a good idea, at least for now. It keeps track of the changes in content items.

**Comment by Piedone** [comment]
Isn't this done, @sarahelsaig?

**Comment by sarahelsaig** [comment]
We have a checkout, but it's a series of MVC actions in a rigid order. I think the original idea of the issue is still valuable, that it should be a collection of workflow activities, to make the whole checkout flow customizable. I've updated the issue title accordingly.
